### PR TITLE
OCD-3151 - Mark (a)(10), (a)(13), (e)(2) 2015 Removed Criteria as Removed and Grey out in UI

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-resources/src/main/resources/jobs.xml [hunk 1]
```diff
@@ -749,6 +749,27 @@
             <durability>true</durability>
             <recover>false</recover>
         </job>
+<<<<<<< HEAD
```
Comment (kekey1) [inline chpl/chpl-resources/src/main/resources/jobs.xml]: Conflict

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java [hunk 2]
```diff
@@ -847,26 +928,34 @@ public ActivityMetadataPage metadataForPendingSurveillances(@RequestParam(requir
...
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.BEARER)
+            })
     @RequestMapping(value = "/acbs/{id}", method = RequestMethod.GET, produces = "application/json; charset=utf-8")
 
     public List<ActivityDetails> activityForACBById(@PathVariable("id") final Long id,
             @RequestParam(required = false) final Long start, @RequestParam(required = false) final Long end)
-                    throws JsonParseException, IOException, EntityRetrievalException, ValidationException {
-        CertificationBodyDTO acb = resourcePermissions.getAcbIfPermissionById(id); // throws 404 if ACB doesn't exist
+            throws JsonParseException, IOException, EntityRetrievalException, ValidationException {
+        CertificationBodyDTO acb = resourcePermissions.getAcbIfPermissionById(id); // throws
```
Comment (andlar) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java:944]: I don't know what setting you have that does things like this to these comments, but maybe consider turning it off? Or updating it to not have changed this? There are a lot of changes to comments in this file just to change spacing and line breaks, and it makes it really hard to figure out what actually changed

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/FeatureList.java [hunk 3]
```diff
@@ -7,6 +7,7 @@ private FeatureList() {
     public static final String CHANGE_REQUEST = "change-request";
     public static final String CONFORMANCE_METHOD = "conformance-method";
     public static final String ENHANCED_UPLOAD = "enhanced-upload";
+    public static final String EFFECTIVE_RULE_DATE_PLUS_18_MONTHS = "effective-rule-date-plus-18-months";
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/FeatureList.java]: Can you move this up a line? It's not alphabetical, which means the PR I'm putting together that has it alphabetized with the other flags will have a merge conflict, or a compilation error, as we're trying to insert the same row in two places

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java (hunk 2 above)
Comment (tmy1313) [reply chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java:944]: I've tried to turn this off...  It only seems to happen to the chpl-api project and not chplk-service.  Made a few morte changes, and we'll see what happens.

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewer.java [hunk 4]
```diff
@@ -99,9 +99,9 @@ private void checkACriteriaHaveRequiredDependencies(CertifiedProductSearchDetail
         errors = validationUtils.checkClassSubsetOfCriteriaForMissingComplementaryCriteriaErrors(A_CRITERIA_NUMBERS_START,
                 attestedCriteria, exceptionsToRequiredByACriteria, exceptionsToACriteria);
         listing.getErrorMessages().addAll(errors);
-        warnings = validationUtils.checkClassSubsetOfCriteriaForMissingComplementaryCriteriaWarnings(A_CRITERIA_NUMBERS_START,
-                attestedCriteria, exceptionsToRequiredByACriteria, exceptionsToACriteria);
-        addListingWarningsByPermission(listing, warnings);
+        //warnings = validationUtils.checkClassSubsetOfCriteriaForMissingComplementaryCriteriaWarnings(A_CRITERIA_NUMBERS_START,
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewer.java]: I think you missed a section of stuff to remove. Github won't let me comment where there is no change, but check a few lines above... around line 95. This Reviewer is used in flexible upload too and I noticed warnings still in there when testing a different feature so came here to check the change.

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/test/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewerTest.java [hunk 5]
```diff
@@ -267,13 +267,13 @@ public void review_a6RemovedCriteriaAttestedWithoutDependencies_asAdmin_hasWarni
...
                 .build();
         reviewer.review(listing);
-        assertEquals(6, listing.getWarningMessages().size());
-        assertTrue(listing.getWarningMessages().contains(String.format(CRITERIA_COMPLEMENT_NOT_FOUND, "170.315 (a)(*)", Util.formatCriteriaNumber(d1))));
-        assertTrue(listing.getWarningMessages().contains(String.format(CRITERIA_COMPLEMENT_NOT_FOUND, "170.315 (a)(*)", Util.formatCriteriaNumber(d2))));
-        assertTrue(listing.getWarningMessages().contains(String.format(CRITERIA_COMPLEMENT_NOT_FOUND, "170.315 (a)(*)", Util.formatCriteriaNumber(d3))));
-        assertTrue(listing.getWarningMessages().contains(String.format(CRITERIA_COMPLEMENT_NOT_FOUND, "170.315 (a)(*)", Util.formatCriteriaNumber(d5))));
-        assertTrue(listing.getWarningMessages().contains(String.format(CRITERIA_COMPLEMENT_NOT_FOUND, "170.315 (a)(*)", Util.formatCriteriaNumber(d6))));
-        assertTrue(listing.getWarningMessages().contains(String.format(CRITERIA_COMPLEMENT_NOT_FOUND, "170.315 (a)(*)", Util.formatCriteriaNumber(d7))));
+        assertEquals(0, listing.getWarningMessages().size());
```
Comment (kekey1) [inline chpl/chpl-service/src/test/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewerTest.java:270]: I think the tests need to be renamed as well (just the suffix "_hasWarnings", to maybe "_noWarnings"). Alternatively, do we still need these tests? Maybe they could be removed. But I think the edit is fine.. just one more layer of checking things.

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewer.java [hunk 6]
```diff
@@ -168,20 +149,12 @@ private void checkE1CriterionHasRequiredDependencies(CertifiedProductSearchDetai
                 criterionService.get(Criteria2015.D_9))
                 .collect(Collectors.toList());
         CertificationCriterion e1 = criterionService.get(Criteria2015.E_1_OLD);
-        CertificationCriterion e1Cures = criterionService.get(Criteria2015.E_1_CURES);
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewer.java]: Why was this code removed?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewer.java [hunk 7]
```diff
@@ -168,20 +149,12 @@ private void checkE1CriterionHasRequiredDependencies(CertifiedProductSearchDetai
...
 
         List<String> errors = validationUtils.checkSpecificCriterionForMissingComplementaryCriteriaErrors(
                 e1,
                 attestedCriteria,
                 requiredByE1Criteria);
         listing.getErrorMessages().addAll(errors);
-
-        errors = validationUtils.checkSpecificCriterionForMissingComplementaryCriteriaErrors(
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredAndRelatedCriteriaReviewer.java]: I also think this should be added back in. If it was removed correctly as part of OCD-3151, then one AQA test needs updated and I can point you to it.

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**